File: Backend/sempi5/src/Domain/SpecializationAggregate/SpecializationMap.cs
namespace Sempi5.Domain.SpecializationAggregate;

public class SpecializationMap
{
    
    public static SpecializationDTO toDTO(Specialization specialization)
    {
        return new SpecializationDTO
        {
            specializationID = specialization.Id.AsLong(),
            specializationName = specialization.specializationName.ToString(),
            specializationCode = specialization.specializationCode.ToString(),
            specializationDescription = specialization.specializationDescription.ToString()
        };
    }
    
    public static Specialization toDomain(SpecializationDTO specializationDTO)
    {
        return new Specialization(
            new SpecializationName(specializationDTO.specializationName),
            new SpecializationCode(specializationDTO.specializationCode),
            new SpecializationDescription(specializationDTO.specializationDescription));
    } 
}

File: SPA/src/app/Mappers/SpecializationMap.ts
import {Specialization} from '../Domain/Specialization';
import {SpecializationDTO} from '../DTO/SpecializationDTO';

export class SpecializationMap {

  public static toDTO(specialization: Specialization): SpecializationDTO {
    return {
      specializationID: specialization.specializationID,
      specializationName: specialization.specializationName
    } as SpecializationDTO;
  }

  public static toDomain(specializationDTO: SpecializationDTO): Specialization {

    return {
      specializationID: specializationDTO.specializationID,
      specializationName: specializationDTO.specializationName,
      specializationCode: specializationDTO.specializationCode,
      specializationDescription: specializationDTO.specializationDescription
    } as Specialization;

  }
}
